frontend/src/app/libs/auth/helpers/createSpy.ts

```diff
@@ -0,0 +1,15 @@
+export function createSpy<T>(prototypeOrMethods: T | Array<keyof T>, params?: { [prop in keyof T]?: T[prop] }): jasmine.SpyObj<T> {
+	const methodsSpy: jasmine.SpyObj<T> =
+		prototypeOrMethods instanceof Array
+			? createSpyObj(prototypeOrMethods)
```

Comment:
This logic looks hard.
Suppose we can create additional method to return methods for createSpyObj.
Like createSpyObj(getMethodsForSpy())

Reply:
Copied from prod code, let's temporary not touch this :)